src/braket/circuits/angled_gate.py

```diff
@@ -41,21 +48,31 @@ def __init__(self, angle: float, qubit_count: Optional[int], ascii_symbols: Sequ
         super().__init__(qubit_count=qubit_count, ascii_symbols=ascii_symbols)
         if angle is None:
             raise ValueError("angle must not be None")
-        self._angle = float(angle)  # explicit casting in case angle is e.g. np.float32
+        if isinstance(angle, FreeParameter):
+            self._angle = angle
+            self._parameterized = True
+            self._parameter = angle
+        else:
+            self._angle = float(angle)  # explicit casting in case angle is e.g. np.float32
+            self._parameterized = False
+            self._parameter = None
 
     @property
-    def angle(self) -> float:
+    def angle(self) -> Union[FreeParameter, float]:
         """
         Returns the angle for the gate
 
         Returns:
-            angle (float): The angle of the gate in radians
+            angle (Union[FreeParameter, float]): The angle of the gate in radians
         """
         return self._angle
 
     def __eq__(self, other):
         if isinstance(other, AngledGate):
-            return self.name == other.name and math.isclose(self.angle, other.angle)
+            if isinstance(self.angle, FreeParameter):
+                return self.name == other.name and self.angle == other.angle
+            else:
+                return self.name == other.name and math.isclose(self.angle, other.angle)
         return NotImplemented
```

Comment:
I know it's not part of the original scope this PR, but might as well fix it; `NotImplemented` is a truthy value, so it should actually be `False`.